recipes-extended/xen/blktap3/OXT-specific-errors-fix.patch

```diff
@@ -0,0 +1,86 @@
+SHORT DESCRIPTION: 
+################################################################################
+Resolve OpenXT specific errors for blktap3.
+
+################################################################################
+LONG DESCRIPTION: 
+################################################################################
+Inclusion of blktap3 into OpenXT requires few changes to be done as below.
+
+The wrappers in mockatests is removed because it requires libcmocka which is not
+provided by any packages. This doesn't impact any functionality as it's a
+testing framework. Since we are aligning with upstream, this removal shouldn't 
+be an issue. However, as and when libcmocka is available it should be included.
+
+################################################################################
+CHANGELOG 
+################################################################################
+Author(s):
+Mahantesh Salimath <salimathm@ainfosec.com>
+
+################################################################################
+REMOVAL 
+################################################################################
+
+################################################################################
+UPSTREAM PLAN
+################################################################################
+No
+
+################################################################################
+INTERNAL DEPENDENCIES 
+################################################################################
+
+################################################################################
+PATCHES 
+################################################################################
+Index: blktap/include/Makefile.am
+===================================================================
+--- blktap.orig/include/Makefile.am
++++ blktap/include/Makefile.am
+@@ -19,6 +19,7 @@ blktap_HEADERS += tap-ctl.h
+ blktap_HEADERS += debug.h
+ blktap_HEADERS += util.h
+ blktap_HEADERS += ../drivers/tapdisk-metrics-stats.h
++blktap_HEADERS += list.h
+ 
+ noinst_HEADERS  = blktap.h
+ noinst_HEADERS += compiler.h
+Index: blktap/include/tap-ctl.h
+===================================================================
+--- blktap.orig/include/tap-ctl.h
++++ blktap/include/tap-ctl.h
+@@ -85,7 +85,7 @@ int tap_ctl_connect_send_receive_ex(int
+ 				    struct timeval *timeout);
+ char *tap_ctl_socket_name(int id);
+ 
+-typedef struct {
++typedef struct tap_list{
+ 	pid_t       pid;
+ 	int         minor;
+ 	int         state;
+Index: blktap/control/Makefile.am
+===================================================================
+--- blktap.orig/control/Makefile.am
++++ blktap/control/Makefile.am
+@@ -9,6 +9,7 @@ AM_CPPFLAGS += -I$(top_srcdir)/include
+ AM_CPPFLAGS += -DTAPDISK_EXEC='"tapdisk"'
+ AM_CPPFLAGS += -DTAPDISK_EXECDIR='"$(libexecdir)"'
+ AM_CPPFLAGS += -DTAPDISK_BUILDDIR='"$(top_builddir)/drivers"'
++AM_CPPFLAGS += -DTAP_CTL_NO_DEFAULT_CGROUP_SLICE
```

Comment:
Can we use bitbake to pass in this flag?

Reply:
Ack, will do